ts/src/connections/tunnelRelayTunnelHost.ts

```diff
@@ -39,17 +46,32 @@ import { SshHelpers } from './sshHelpers';
 import { MultiModeTunnelHost } from './multiModeTunnelHost';
 import { TunnelHostBase } from './tunnelHostBase';
 import { tunnelRelaySessionClass } from './tunnelRelaySessionClass';
+import { SessionPortKey } from './sessionPortKey';
+import { PortRelayConnectRequestMessage } from './messages/portRelayConnectRequestMessage';
+import { PortRelayConnectResponseMessage } from './messages/portRelayConnectResponseMessage';
 
 const webSocketSubProtocol = 'tunnel-relay-host';
+const webSocketSubProtocolv2 = 'tunnel-relay-host-v2-dev';
+
+// Check for an environment variable to determine which protocol version to use.
+// By default, prefer V2 and fall back to V1.
+const protocolVersion = process?.env && process.env.DEVTUNNELS_PROTOCOL_VERSION;
+const connectionProtocols =
+    protocolVersion === '1' ? [webSocketSubProtocol] :
+    protocolVersion === '2' ? [webSocketSubProtocolv2] :
+    [webSocketSubProtocolv2, webSocketSubProtocol];
 
 /**
  * Tunnel host implementation that uses data-plane relay
  *  to accept client connections.
  */
 export class TunnelRelayTunnelHost extends tunnelRelaySessionClass(
     TunnelHostBase,
-    webSocketSubProtocol,
+    connectionProtocols,
 ) {
+    public static readonly webSocketSubProtocol = webSocketSubProtocol;
```

Comment:
Can these two statics go to `tunnelRelaySessionClass`? This way they'll be shared with `TunnelRelayTunnelClient`.

Reply:
The client and host protocol names are not the same, so they can't be shared in that way.